canonical-websites/snapcraft.io
# Show categories on the store page

# Done

Fixes https://github.com/canonical-websites/snapcraft.io/issues/733

- Added extra call on the backend to get the list of categories from the API
- Modified the search API to include the category
- Updated templates to show the list of categories
- Added JS to fetch snaps for the listed categories
- Added JS `Carousel` class (generic so we can replace the screenshots one with it) for scrolling through category snaps
- Added extra sass component

# QA

- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004/store
- Make sure the categories load
- Click a category heading
- Make sure you get the list of snaps in that category (search results page, for now)
- Paginate through
- Use the search input to search for a snap name that exists
- Results should show, as expected
- Paginate through
- Search for a snap that doesn't exist
- Error message should show followed by the categories
- Clear the text search and select a category from the dropdown
- Click search
- Category should show
- Search for the name of a snap in that category
- The snap should show
- Search for a snap that doesn't exist
- Error message should display with the carousel for the selected category

# Screenshots
## Store page
![screenshot-2018-6-28 install linux apps in seconds snap store 1](https://user-images.githubusercontent.com/479384/42046894-c8e686b4-7af6-11e8-90c5-dec64e1f4f42.png)
## Category page
![screenshot-2018-6-28 games linux software in the snap store](https://user-images.githubusercontent.com/479384/42046934-dec6833a-7af6-11e8-94aa-3deb64210420.png)
## Search results (All snaps)
![screenshot-2018-6-28 snap search results for 1](https://user-images.githubusercontent.com/479384/42047005-05a9ab12-7af7-11e8-986f-d347fdf0e1b0.png)
## Search results (Games)
![screenshot-2018-6-28 snap search results for 2](https://user-images.githubusercontent.com/479384/42047051-257e849e-7af7-11e8-813d-fbbf9a74dd97.png)
## No results (All snaps)
![screenshot-2018-6-28 snap search results for 3](https://user-images.githubusercontent.com/479384/42047079-3961cce6-7af7-11e8-821d-7d93f3c16082.png)
## No results (Games)
![screenshot-2018-6-28 snap search results for 4](https://user-images.githubusercontent.com/479384/42047100-47061820-7af7-11e8-9646-a50b7df21653.png)
